src/backend/glutin.rs

```diff
@@ -0,0 +1,359 @@
+    }
+}
+
+/// Errors that may happen when driving the event loop of `GlutinInputBackend`
+pub enum GlutinInputError
+{
+    /// The underlying `glutin` `Window` was closed. No further events can be processed.
+    ///
+    /// See `GlutinInputBackend::process_new_events`.
+    WindowClosed
+}
+
+/// Abstracted event loop of a `glutin` `Window` implementing the `InputBackend` trait
+///
+/// You need to call `process_new_events` periodically to receive any events.
+pub struct GlutinInputBackend
+{
+    window: Rc<Window>,
+    time_counter: u32,
+    seat: Seat,
+    handler: Option<Box<InputHandler + 'static>>,
+}
+
+impl InputBackend for GlutinInputBackend
+{
+    fn set_handler<H: InputHandler + 'static>(&mut self, mut handler: H) {
+        if self.handler.is_some() {
+            self.clear_handler();
+        }
+        handler.on_seat_created(&self.seat);
+        self.handler = Some(Box::new(handler));
+    }
+
+    fn get_handler(&mut self) -> Option<&mut InputHandler>
+    {
+        self.handler.as_mut().map(|handler| handler as &mut InputHandler)
+    }
+
+    fn clear_handler(&mut self) {
+        if let Some(ref mut handler) = self.handler {
+            handler.on_seat_destroyed(&self.seat);
+        }
+        self.handler = None;
+    }
+
+    fn set_cursor_position(&mut self, x: u32, y: u32) -> Result<(), ()> {
+        if let Some((win_x, win_y)) = self.window.get_position() {
+            self.window.set_cursor_position(win_x + x as i32, win_y + y as i32)
+        } else {
+            Err(())
+        }
+    }
+}
+
+impl GlutinInputBackend
+{
+    fn new(window: Rc<Window>) -> GlutinInputBackend
+    {
+        GlutinInputBackend {
+            window: window,
+            time_counter: 0,
+            seat: Seat::new(0),
+            handler: None,
+        }
+    }
+
+    /// Processes new events of the underlying event loop to drive the set `InputHandler`.
+    ///
+    /// You need to periodically call this function to keep the underlying event loop and
+    /// `Window` active. Otherwise the window may no respond to user interaction and no
+    /// input events will be received by a set `InputHandler`.
+    ///
+    /// Returns an error if the `Window` the window has been closed. Calling
+    /// `process_new_events` again after the `Window` has been closed is considered an
+    /// application error and unspecified baviour may occur.
+    ///
+    /// The linked `GlutinWindowedRenderer` will error with a lost Context and should
+    /// not be used anymore as well.
+    pub fn process_new_events(&mut self) -> Result<(), GlutinInputError>
+    {
+        for event in self.window.poll_events()
+        {
+            if let Some(ref mut handler) = self.handler {
+                match event {
+                    Event::KeyboardInput(state, key_code/*TODO: Is this really the keycode? glutins docs don't tell*/, _) => handler.on_keyboard_key(&self.seat, self.time_counter, key_code as u32, state.into(), 1),
```

Comment:
For the wayland backend, of glutin, this is the raw keycode (truncated from `u16` to `u8`). It's probably the same for others, afaik.

Also, this `1` means we do not track simultaneous key presses? See my comment on the definition of the trait.